src/routes/auth.ts

```diff
@@ -0,0 +1,96 @@
+import express from 'express'
+import { ErrorTypes, SiweMessage } from 'siwe'
+import { validateSchema } from '../schema'
+import {
+  AuthRequestBody,
+  InvalidSiweParamsError,
+  NotImplementedError,
+} from '../types'
+import { asyncRoute } from './async-route'
+
+const MAX_EXPIRATION_TIME_MS = 4 * 60 * 60 * 1000 // 4 hours
+const VERSION = '1'
+const CHAIN_ID = 42220
+
+function validateNonce(_nonce: string) {
+  throw new NotImplementedError('Nonce validation not implemented')
+}
+
+function validateExpirationTime(issuedAt: string, expirationTime?: string) {
+  if (!expirationTime) {
+    throw new InvalidSiweParamsError('Missing expiration time')
+  }
+  const issuedAtDate = new Date(issuedAt)
```

Comment:
issued-at must be in the past, see [here](https://github.com/fiatconnect/specification/pull/17/files#diff-07ebdd7905df816d952f03b3934460f07609cb2db493130f09a81869287ab9a9R548)

Reply:
same with expiration-time

Reply:
Looks like the siwe package ensures expiration-time is in the future, but doesn't hurt double checking here.